jupyter_client/utils.py

```diff
@@ -18,7 +18,12 @@ def wrapped(*args, **kwargs):
         import nest_asyncio  # type: ignore
 
         nest_asyncio.apply(loop)
-        return loop.run_until_complete(coro(*args, **kwargs))
+        future = asyncio.ensure_future(coro(*args, **kwargs))
+        try:
+            return loop.run_until_complete(future)
+        except KeyboardInterrupt:
+            future.cancel()
+            raise
```

Comment:
Maybe we want to handle any kind of exception?

```suggestion
        except BaseException as e:
            future.cancel()
            raise e
```

Reply:
I agree with your suggestion.